src/Form/LegacyConsumer/resources/js/conditinal-fields.js

```diff
@@ -0,0 +1,193 @@
+			}
+		}
+	}
+
+	/**
+	 * Setup state for condition visibility settings.
+	 * state contains list of watched elements per donation form.
+	 *
+	 * @unreleased
+	 */
+	function addVisibilityConditionsToStateForDonationForm(donationForm) {
+		const uniqueDonationFormId = donationForm.getAttribute('data-id');
+		const watchedFields = getWatchedElementNames(donationForm);
+
+		// Add donation form to state only if visibility conditions exiting for at least form field.
+		if (uniqueDonationFormId && Object.keys(watchedFields).length) {
+			state[uniqueDonationFormId] = watchedFields;
+		}
+	}
+
+	/**
+	 * Setup state for condition visibility settings.
+	 * state contains list of watched elements per donation form.
+	 *
+	 * @unreleased
+	 */
+	async function addVisibilityConditionsToStateForAllDonationForm() {
+		await document.querySelectorAll('form.give-form').forEach(addVisibilityConditionsToStateForDonationForm);
+	}
+
+	/**
+	 * @unreleased
+	 */
+	function applyVisibilityConditionsAttachedToWatchedField(donationForm, fieldName) {
+		const uniqueDonationFormId = donationForm.getAttribute('data-id');
+
+		if (
+			donationForm &&
+			uniqueDonationFormId &&
+			(uniqueDonationFormId in state)
+		) {
+			const formState = state[uniqueDonationFormId];
+
+			if (fieldName in formState) {
+				handleVisibility(donationForm, formState[fieldName])
+			}
+		}
+	}
+
+	/**
+	 * @unreleased
+	 * @param donationForm
+	 */
+	function applyVisibilityConditionsToDonationForm(donationForm) {
+		const uniqueDonationFormId = donationForm.getAttribute('data-id');
+
+		if (uniqueDonationFormId && (uniqueDonationFormId in state)) {
+			const formState = state[uniqueDonationFormId];
+
+			for (const [watchedFieldName, visibilityConditions] of Object.entries(formState)) {
+				handleVisibility(
+					document.querySelector(`form[data-id="${uniqueDonationFormId}"]`)
+						.closest('.give-form'),
+					visibilityConditions
+				);
+			}
+		}
+	}
+
+	/**
+	 * @unreleased
+	 */
+	function applyVisibilityConditionsToAllDonationForm() {
+		for (const [uniqueDonationFormId, donationFormState] of Object.entries(state)) {
+			for (const [watchedFieldName, visibilityConditions] of Object.entries(donationFormState)) {
+				handleVisibility(
+					document.querySelector(`form[data-id="${uniqueDonationFormId}"]`)
+						.closest('.give-form'),
+					visibilityConditions
+				);
+			}
+		}
+	}
+
+	addVisibilityConditionsToStateForAllDonationForm()
+		.then(
+			r => {
+				applyVisibilityConditionsToAllDonationForm();
+
+				// Apply visibility conditions to donation form when donor switch gateway.
+				document.addEventListener(
+					'give_gateway_loaded',
+					event => {
+						const donationForm = document.getElementById(event.detail.formIdAttribute);
+						addVisibilityConditionsToStateForDonationForm(donationForm);
+						applyVisibilityConditionsToDonationForm(donationForm);
+					}
+				);
+
+
+				// Look for change in watched elements.
+				document.addEventListener(
```

Comment:
If you ask me, maybe it will be better to add some kind of identification ( class, attribute, etc. ) inside the consumer that will help you identify all inputs that are handling other fields visibility ( we use that approach in P2P ). Attaching change events on each input is not very performant.

Reply:
fixed: https://github.com/impress-org/givewp/compare/72ed8c4845c6...cce2a1f53929

Reply:
code refactored: https://github.com/impress-org/givewp/compare/cce2a1f53929...2763150e4b90